jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -47,6 +53,37 @@
         return children;
     }
 
+    @SuppressWarnings("unchecked")
+    public static boolean updateFilters(List<Object> arguments, IProgressMonitor monitor) throws Exception {
+        final List<String> patterns = (List<String>) arguments.get(0);
+        final IProject[] projects = getWorkspaceRoot().getProjects();
+        final Set<IProject> filterUpdated = new HashSet<>();
+        for (final IProject project : projects) {
+            if (!project.exists()) {
+                continue;
+            }
+            final Set<String> filters = Arrays.stream(project.getFilters())
+                .map(filter -> (String) filter.getFileInfoMatcherDescription().getArguments())
+                .collect(Collectors.toSet());
+            for (final String pattern: patterns) {
+                if (filters.contains(pattern)) {
+                    continue;
+                }
+                project.createFilter(
+                    IResourceFilterDescription.EXCLUDE_ALL |
+                    IResourceFilterDescription.FILES |
+                    IResourceFilterDescription.FOLDERS |
+                    IResourceFilterDescription.INHERITABLE,
+                    new FileInfoMatcherDescription("org.eclipse.core.resources.regexFilterMatcher", pattern), 0, monitor);
+                filterUpdated.add(project);
```

Comment:
The project won't delete the filter when corresponding exclude settings got removed.

Reply:
Has been solved in new commit, which will delete all the filters which is not in the pattern list provided by client.